### DISCUSSION TITLE
Composing Store make typescript type unwrap and different

### DISCUSSION THREAD

**Discussion by defghy** [vuejs/pinia#2665] [Help and Questions]
In defineStore composing function, type is right
![image](https://github.com/vuejs/pinia/assets/5386958/a39c6b3c-2692-400a-a54b-3d62aca00032)

The ReturnValue of defineStore(), type transfer to specific object, which is not right
![image](https://github.com/vuejs/pinia/assets/5386958/fe44857e-c4f1-4d4c-9c9c-7001540f33c3)

I cannot use the object as `BoardData`
![image](https://github.com/vuejs/pinia/assets/5386958/53a86862-70c2-4b5a-a31d-16c96c553f99)

`BoardData` transfer to specific `{ boardId: ''...}`

**Comment by defghy** [comment]
In some research `pinia.d.ts`，this is difference between `UnwrapRef` and `ShallowUnwrapRef` in 'vue'

`UnwrapRef`
![image](https://github.com/vuejs/pinia/assets/5386958/e00bd90a-c9b9-47e2-bf68-3761e40cd890)

`ShallowUnwrapRef`
![image](https://github.com/vuejs/pinia/assets/5386958/10397054-eafa-4885-a64d-03b489599f81)

Maybe `ShallowUnwrapRef` is more reasonable here?

source d.ts

```

(pinia?: Pinia | null | undefined, hot?: StoreGeneric): Store<Id, S, G, A>;

export declare type Store<Id extends string = string, S extends StateTree = {}, G = {}, A = {}> = _StoreWithState<Id, S, G, A> & UnwrapRef<S> & _StoreWithGetters<G> & (_ActionsTree extends A ? {} : A) & PiniaCustomProperties<Id, S, G, A> & PiniaCustomStateProperties<S>;
```

**Comment by kevinramharak** [comment]
Can you use the playground to provide an example:
https://play.pinia.vuejs.org/#eNqNU8Fu2zAM/RXBl6RoIrXI0AFZ0mUbeugO27DuaAxwbCZVKlGCJKcGAv/7aClx3G4o5pNEPj4+Un6H7JO1fF9DNs8WvnTSBuYh1JapArfLPAs+z25zlNoaF9iB1R4egnHAWrZxRrMRF6WpMYDjwY9yzLE06IkkgpY9fnzR5WKU44cchWD0/f6Y40KkvtSFLgG0VUUAujG2WNchGGSrUsnyidQc6y8vSdQ9lg40IKk6pHYcWdsuRCoigoUYsGWTLA0x1YXlO2+QRj50XfJjgiadsxjpYlaiLLpInol45uD1dO3Ms6dZdwSenKAr2p+oYB+MUX5aWJnKHkOwfi5EWSHhK1By7zhCEGi1+KtmdcNv+LVQci2okZBYQfOyDVVMK9D/w36Crq749Tv+PrLuZ0dS3bF2pG2OLa0leHqyjdy+WkpptJUK3HcbJD3pi+UUSpnnrzEWXA29xPIRyqd/xHe+Sap/OKD17WEwVijcFkJK3z18g4bOfVKbqlaEfiP5E7xRdacxwT7XWJHsAS6qvY9vLHH7y981AdCfhuqExm1EfFzzlzdGP8ud8dlgi2cX0Ap7u1SwkfjKMfFvIqv0IAebPkndo4ugiclkpt50yyHheHTsOZqw8QVb3iaRqQQJS7zjK/IdY2e3dTdHBndIjckvJJ4QWfsHsDpZqA==

You can use `// ^?` to have the editor show the type as an inlay hint as provided in the link.
